### PR TITLE
RRv4 checks and navbar components added to App

### PR DESCRIPTION
**Pull request by mmahalwy**
### Title of change
PR number two for shaping up React React 4

- Fixes the checks for valid chapter/verse
- Adds navbars to the routes

### Checklist

- [ ] Unit tests written
- [x] Manually tested
- [x] New dependencies were added with `yarn`
- [x] Prettier & ESLint were run

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32887

**Comment by naveed-ahmad** [comment]
@mmahalwy manually tested and there are lot of breaks:

http://staging.quran.com:32887/1:1 => Should redirect to /1/1
http://staging.quran.com:32887/001  => Should redirect to /1 but got 500 error
http://staging.quran.com:32887/1:120 => Should redirect to invalid-ayah 

Jump to Surah or Ayah is breaking on Surah page.

**Comment by mmahalwy** [comment]
@naveed-ahmad okay, good to know. Thanks for doing this. I will add tests and fix these

File: src/routes.js [hunk 1]
```diff
@@ -114,41 +128,49 @@ export const routes = [
...
-        render={(routeProps) => {
+        render={({ staticContext, ...routeProps }) => {
+          if (staticContext) {
+            const contextFunction = setContext || defaultSetContext;
+
+            Object.assign(staticContext, contextFunction(staticContext));
+          }
+
           if (__CLIENT__) {
             routePromises({
```
Comment (sabeurthabti) [inline src/routes.js]: can we catch this promise and return an error page?

Comment (mmahalwy) [reply src/routes.js]: Good catch. I will add it inshallah

**Comment by naveed-ahmad** [comment]
rebuild

**Comment by mmahalwy** [comment]
rebuild

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32892

**Comment by mmahalwy** [comment]
Going to merge this. Iterating on it on staging

**Pull request merged**